core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/aggregation/GroupByReceiverImpl.kt

```diff
@@ -118,7 +119,19 @@ internal class GroupByReceiverImpl<T>(override val df: DataFrame<T>, override va
                 pivot.aggregator.values.clear()
             }
 
-            is AggregateInternalDsl<*> -> yield(value.copy(value = value.value.df))
+            is AggregateInternalDsl<*> -> {
+                // Attempt to create DataFrame<Type> from AggregateInternalDsl<Type>
+                val dfType = value.type?.arguments?.firstOrNull()?.type
```

Comment:
I'm a little bit worried about this expression with 6 ? - what if something goes wrong and will be null, I could not explain myself the logic, should we handle the alternative (or common alternative)?

Reply:
That's what the `?:` is for at the end. We try to create `DataFrame<Type1>` from `AggregateInternalDsl<Type1>`. However, in theory, any KType can be supplied by the user. KTypes like `GroupByReceiverImpl<Type1>` , `DataFrame<Type1>` (which would both still work), but unfortunately also KTypes like `CompletelyCustomType` without arguments. So in cases where this type argument cannot be fetched, we default to `DataFrame<*>` with the `?:`.